hw/ip/prim/dv/prim_lfsr/prim_lfsr_sim_cfg.hjson

```diff
@@ -37,6 +37,11 @@
     }
   ]
 
+  // TODO: each build_mode needs to supply {build_mode}_vcs_cov_hier, else the
```

Comment:
@sriyerg do we really want each module hjson to supply something vendor-specific? 
this seems like the wrong way to go.
it should at least be dependent on what simulator is used.

or am I missing a point here?

Reply:
I am not sure I understand your question. What is "vendor-specific" here? 
Are you pointing out the fact that 'vcs' is in the key name and we should avoid it? We  cannot since it is specific to VCS.

Reply:
Ah, is this not required for other simulators? I think Rasmus is wondering whether we'll end up needing "(num simulators) times (num options)" entries.

Reply:
@sriyerg sorry for not being more clear.
but Yes as @rswarbrick suggests I am worried about ending up with too many entries, not so much the name.

Reply:
We should be able to run `prim_lfsr` with Xcelium without specifying Xcelium specific params as long as we do not enable coverage. We do not need to replicate this for ALL simulators since we are signing off with only one simulator (we enable coverage only if we are signing off with that tool). 
 
Plus this only affects `prim_lfsr` and none other. If the complete regression for an IP can be run with just the two common build modes that exist today (default, cover_reg_top), then nothing extra needs to be added to the IP's sim cfg hjson - they would be using the ones already set in `common_sim_cfg.hjson`. These options need to be set *only* if there are tests that use `build_modes` different from {default, cover_reg_top}.

This is a side-effect of the changes introduced in this PR to allow build-mode-specific coverage options to be set in a very flexible way. Note the TODO above - I put it in there because its on my mind to address it (I already agree with the point you make!). For the time being, the value this PR adds far outweighs this minor inconvenience.